client/valor/schemas/symbolic/collections.py

```diff
@@ -0,0 +1,798 @@
+import typing
+from typing import Any, Optional
+
+from valor.enums import TaskType
+from valor.schemas.symbolic.types import (
+    Bool,
+    BoundingBox,
+    BoundingPolygon,
+    Date,
+    DateTime,
+    Duration,
+    Embedding,
+    Equatable,
+    Float,
+    Integer,
+    LineString,
+    MultiLineString,
+    MultiPoint,
+    MultiPolygon,
+    Nullable,
+    Point,
+    Polygon,
+    Raster,
+    Score,
+    String,
+    Symbol,
+    TaskTypeEnum,
+    Time,
+    Variable,
+)
+
+
+def _get_atomic_type_by_value(other: Any):
+    """Retrieves variable type using built-in type."""
+    if Bool.supports(other):
+        return Bool
+    elif String.supports(other):
+        return String
+    elif Integer.supports(other):
+        return Integer
+    elif Float.supports(other):
+        return Float
+    elif DateTime.supports(other):
+        return DateTime
+    elif Date.supports(other):
+        return Date
+    elif Time.supports(other):
+        return Time
+    elif Duration.supports(other):
+        return Duration
+    elif MultiPolygon.supports(other):
+        return MultiPolygon
+    elif Polygon.supports(other):
+        return Polygon
+    elif MultiLineString.supports(other):
+        return MultiLineString
+    elif LineString.supports(other):
+        return LineString
+    elif MultiPoint.supports(other):
+        return MultiPoint
+    elif Point.supports(other):
+        return Point
+    else:
+        raise NotImplementedError(str(type(other).__name__))
+
+
+def _get_atomic_type_by_name(name: str):
```

Comment:
nit: using a dictionary lookup is probably faster and more pythonic than using long if/else chains